chromium_src/chrome/browser/notifications/notification_handler.h

```diff
@@ -10,33 +10,6 @@
 #include "src/chrome/browser/notifications/notification_handler.h"
 #undef NotificationHandler
 
-#include "base/check.h"
-
-class NotificationHandler : public NotificationHandler_ChromiumImpl {
- public:
-  // GENERATED_JAVA_ENUM_PACKAGE: org.chromium.chrome.browser.notifications
-  // GENERATED_JAVA_CLASS_NAME_OVERRIDE: NotificationType
-  enum class Type {
-    WEB_PERSISTENT = 0,
-    WEB_NON_PERSISTENT = 1,
-    EXTENSION = 2,
-    SEND_TAB_TO_SELF = 3,
-    TRANSIENT = 4,
-    PERMISSION_REQUEST = 5,
-    SHARING = 6,
-    ANNOUNCEMENT = 7,
-    NEARBY_SHARE = 8,
-    BRAVE_ADS = 9,
-    NOTIFICATIONS_MUTED = 10,
-    TAILORED_SECURITY = 11,
-    MAX = TAILORED_SECURITY,
-  };
-
-  // Make sure we know if the original enum gets changed.
-  NotificationHandler() {
-    DCHECK(static_cast<int>(NotificationHandler_ChromiumImpl::Type::MAX) + 1 ==
-           static_cast<int>(Type::TAILORED_SECURITY));
-  }
-};
+#include "chrome/browser/notifications/notification_handler_impl.h
```

Comment:
Is this new header file introduced because we have new `NotificationHandler` class as upstream one is renamed to `NotificationHandler_ChromiumImpl`?

Reply:
it's to fix a dep issue with https://github.com/brave/brave-core/pull/16466/files#diff-d83694c57412009f4a8044d65b6903842e59c9b0c89c178c7044e6bf244a38b8R11
```
ERROR at //chrome/browser/notifications/stub_notification_display_service.cc:11:11: Can't include this header from here.
#include "chrome/browser/notifications/notification_handler.h"
          ^--------------------------------------------------
The target:
  //chrome/browser:test_support
is including a file from the target:
  //brave/android:brave_android_java_enums_srcjar
```
and there's no gn target for it to go into. This is really a temporary workaround to a bigger problem with this override

Reply:
actually it could go in brave_chrome_browser_sources since notification_handler.h is in chrome/browser